cpp/modmesh/toggle/RadixTree.hpp

```diff
@@ -68,66 +68,59 @@ class TimedEntry
 template <typename T>
 class RadixTreeNode
 {
-private:
-    using ChildList = std::list<std::unique_ptr<RadixTreeNode>>;
-
 public:
-    RadixTreeNode()
-        : prev(nullptr){};
-    RadixTreeNode(std::string name, int key)
-        : m_name(std::move(name))
+
+    using child_list_type = std::list<std::unique_ptr<RadixTreeNode<T>>>;
```

Comment:
Use snake_case for internal type names.  Also add back the explicit `<T>` in `RadixTreeNode<T>`.

Reply:
Thanks for helping adding back explicit T, which I fully forgot.
But will there be any negative scenarios for aliasing without passing template parameter in this case?

Reply:
It reads strange.  Semantically it's the same with or without the template argument, IIRC.  But one needs to consult the standard to be sure which is right.  It hurts productivity.  When in doubt, choose the least surprising way to write it.